web_poet/testing/pytest.py

```diff
@@ -124,12 +132,29 @@ def _format_extra_fields(self, extra_fields):
         return "\n".join(lines)
 
 
+class WebPoetNoToItemException(_WebPoetItem):
+    def runtest(self) -> None:
+        self.fixture.assert_no_toitem_exceptions()
+
+    def reportinfo(self):
+        return (
+            self._path,
+            0,
+            f"{self.fixture.short_name}: to_item doesn't raise an error",
+        )
+
+
 class WebPoetFieldItem(_WebPoetItem):
     def __init__(self, *, field_name: str, **kwargs) -> None:
         super().__init__(**kwargs)
         self.field_name = field_name
 
     def runtest(self) -> None:
+        if self.fixture.to_item_raised():
+            raise pytest.skip(
+                f"Skipping a test for item.{self.field_name} "
+                f"because to_item raised an exception"
+            )
```

Comment:
I am unsure about this one. Shouldn’t checks for fields that do not raise an exception still pass?

Reply:
We can't get the final values for other fields if one of the fields fail. In this case to_item raises an error, so there are no values for other fields. 

I think that using page.field output directly is not the way to go, because we want to support page objects which don't use field decorator.

Reply:
> we want to support page objects which don't use field decorator

:+1: